backend/src/schema/resolvers/postsInGroups.spec.js

```diff
@@ -0,0 +1,1495 @@
+                  title: 'A post to a public group',
+                  content: 'I am posting into a public group as a member of the group',
+                },
+              ]),
+            },
+            errors: undefined,
+          })
+        })
+
+        it('shows post of a closed group', async () => {
+          await expect(
+            query({ query: postQuery(), variables: { id: 'post-to-closed-group' } }),
+          ).resolves.toMatchObject({
+            data: {
+              Post: expect.arrayContaining([
+                {
+                  id: 'post-to-closed-group',
+                  title: 'A post to a closed group',
+                  content: 'I am posting into a closed group as a member of the group',
+                },
+              ]),
+            },
+            errors: undefined,
+          })
+        })
+
+        it('shows post of a hidden group', async () => {
+          await expect(
+            query({ query: postQuery(), variables: { id: 'post-to-hidden-group' } }),
+          ).resolves.toMatchObject({
+            data: {
+              Post: expect.arrayContaining([
+                {
+                  id: 'post-to-hidden-group',
+                  title: 'A post to a hidden group',
+                  content: 'I am posting into a hidden group as a member of the group',
+                },
+              ]),
+            },
+            errors: undefined,
+          })
+        })
+      })
+    })
+
+    describe('filter posts', () => {
+      beforeAll(async () => {
+        authenticatedUser = newUser
+        await mutate({
+          mutation: createPostMutation(),
+          variables: {
+            id: 'post-without-group',
+            title: 'A post without a group',
+            content: 'As a new user, I do not belong to a group yet.',
+          },
+        })
+      })
+
+      describe('without authentication', () => {
+        beforeEach(async () => {
+          authenticatedUser = null
+        })
+
+        it('shows a the post of the public group and the post without group', async () => {
+          const result = await query({ query: filterPosts(), variables: {} })
+          expect(result.data.Post).toHaveLength(2)
+          expect(result).toMatchObject({
+            data: {
+              Post: expect.arrayContaining([
+                {
+                  id: 'post-to-public-group',
+                  title: 'A post to a public group',
+                  content: 'I am posting into a public group as a member of the group',
+                },
+                {
+                  id: 'post-without-group',
+                  title: 'A post without a group',
+                  content: 'As a new user, I do not belong to a group yet.',
+                },
+              ]),
+            },
+            errors: undefined,
+          })
+        })
+      })
+
+      describe('as new user', () => {
+        beforeEach(async () => {
+          authenticatedUser = newUser
```

Comment:
So here the `newUser` is not new to its own created post `post-without-group`. Therefore this name is misleading.
`anyUser` may be better to create this post.

Reply:
What comes into my eyes is as well:

Why here not:
```
authenticatedUser = await newUser.toJson()
```
like with all the others?
See similar places to be changed with `newUser` above and below …

Ah I see, because the creation was by `SignupVerification` and not by the factory.

Can we name `newUser` then a bit different to distinguish that different behavior?
Maybe `newUserSignup` or `newUserResult` or … ?

Reply:
This is not working. New User has a signup verification via the API. All others are seeded by neode.

Reply:
Have you read mine till the end?
That's what I got myself aware of and therefore I suggested something else …

Reply:
I keep the name of new user.